lib/mayaUsd/render/vp2RenderDelegate/material.cpp

```diff
@@ -2580,20 +2592,23 @@ void HdVP2Material::_UpdateLoadedTexture(
     // function on idle to delete the task object.
     _textureLoadingTasks.erase(path);
 
-    // Check the local cache again, do not overwrite if same texture has
-    // been loaded asynchronously
-    if (_textureMap.find(path) != _textureMap.end()) {
-        return;
```

Comment:
The early exit here is wrong. It also stops marking the sprim dirty and triggering a refresh.